modules/porous_flow/include/materials/PorousFlowSingleComponentFluid.h

```diff
@@ -30,41 +30,50 @@ class PorousFlowSingleComponentFluid : public PorousFlowFluidPropertiesBase
   virtual void initQpStatefulProperties() override;
   virtual void computeQpProperties() override;
 
+  /// If true, this Material will compute density and viscosity, and their derivatives
+  const bool _compute_rho_nu;
+
+  /// If true, this Material will compute internal energy and its derivatives
+  const bool _compute_internal_energy;
+
+  /// If true, this Material will compute enthalpy and its derivatives
+  const bool _compute_enthalpy;
+
```

Comment:
OK.  I'm hoping that users will "never" have to use these flags.  hopefully they can be set appropriately with the actions.